llvm/tools/sycl-post-link/SpecConstants.cpp

```diff
@@ -332,13 +333,14 @@ Instruction *emitCall(Type *RetTy, StringRef BaseFunctionName,
 }
 
 Instruction *emitSpecConstant(unsigned NumericID, Type *Ty,
-                              Instruction *InsertBefore) {
+                              Instruction *InsertBefore,
+                              Constant *DefaultValue) {
   Function *F = InsertBefore->getFunction();
   // Generate arguments needed by the SPIRV version of the intrinsic
   // - integer constant ID:
   Value *ID = ConstantInt::get(Type::getInt32Ty(F->getContext()), NumericID);
   // - default value:
-  Value *Def = getDefaultCPPValue(Ty);
+  Value *Def = DefaultValue != nullptr ? DefaultValue : getDefaultCPPValue(Ty);
```

Comment:
Shouldn't it work simpler way?
```suggestion
  Value *Def = DefaultValue ? DefaultValue : getDefaultCPPValue(Ty);
```

Reply:
> Shouldn't it work simpler way?

Sure, this is a left-over from some debugging

Reply:
Fixed in a5902e417a27d1338cee6487cdb35081f87e0601